api/shared/compare/mixins.py

```diff
@@ -100,3 +142,37 @@ def flags(self, request, *args, **kwargs):
             for flag_name in comparison.non_carried_forward_flags
         ]
         return Response(FlagComparisonSerializer(flags, many=True).data)
+
+    @action(detail=False, methods=["get"])
+    @torngit_safe
+    def impacted_files(self, request, *args, **kwargs):
+        comparison = self.get_object()
+
+        return Response(
```

Comment:
Alternatively maybe the serializer includes the state and we update the docs (the docstring comment in the child view) to explain that this state might change after the initial request.

Reply:
I see, right now if the comparison is still being computed it would return an empty object for the impacted files. I think the most consistent thing to do during pending state is to compute on-the-fly the files similar to what's already happening when the `CommitComparison` object doesn't exist yet.